backend/src/helpers/secrets.ts

```diff
@@ -833,26 +817,27 @@ export const deleteSecretHelper = async ({
   environment,
   type,
   authData,
-  secretPath = "/",
+  secretPath = "/"
 }: DeleteSecretParams) => {
   const secretBlindIndex = await generateSecretBlindIndexHelper({
     secretName,
-    workspaceId: new Types.ObjectId(workspaceId),
+    workspaceId: new Types.ObjectId(workspaceId)
   });
 
   // if using service token filter towards the folderId by secretpath
   if (authData.authPayload instanceof ServiceTokenData) {
-    const { secretPath: serviceTkScopedSecretPath } = authData.authPayload;
+    const { scopes: tkScopes } = authData.authPayload;
+    const validScope = tkScopes.find(
+      (scope) =>
+        picomatch.isMatch(secretPath, scope.secretPath, { strictSlashes: false }) &&
+        scope.environment === environment
+    );
```

Comment:
I see that we do a lot of repeating of the same few lines. Is it possible to make it into a function or is there a reason you didn't put it into a function?

```
const findValidScope = (authPayload, secretPath, environment) => {
  const { scopes: tkScopes } = authPayload;
  const validScope = tkScopes.find(
    (scope) =>
      picomatch.isMatch(secretPath, scope.secretPath, { strictSlashes: false }) &&
      scope.environment === environment
  );
  return validScope;
};
```

Reply:
Done